bats/keyless-tables.bats

```diff
@@ -0,0 +1,369 @@
+c0,c1
+0,0
+2,2
+1,1
+1,1
+CSV
+    dolt table import -c tbl data.csv
+    run dolt sql -q "SELECT count(*) FROM tbl;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "4" ]] || false
+    run dolt sql -q "SELECT * FROM tbl;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "keyless table update" {
+    cat <<CSV > data.csv
+c0,c1
+0,0
+2,2
+1,1
+1,1
+CSV
+    # update is a pure append
+    dolt table import -u keyless data.csv
+    run dolt sql -q "SELECT count(*) FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "8" ]] || false
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "keyless table replace" {
+    cat <<CSV > data.csv
+c0,c1
+7,7
+8,8
+9,9
+CSV
+    dolt table import -r keyless data.csv
+    run dolt sql -q "SELECT count(*) FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "3" ]] || false
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "9,9" ]] || false
+
+}
+
+@test "keyless diff against working set" {
+    dolt sql -q "INSERT INTO keyless VALUES (9,9);"
+    run dolt diff
+    [ $status -eq 0 ]
+    [[ "$output" = "|  +  | 9  | 9  |" ]] || false
+}
+
+@test "keyless merge fast-forward" {
+    dolt checkout -b other
+    dolt sql -q "INSERT INTO keyless VALUES (9,9);"
+    dolt commit -am "9,9"
+    dolt checkout master
+    run dolt merge other
+    [ $status -eq 0 ]
+    run dolt sql -q "SELECT * FROM keyless WHERE c0 > 6;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "9,9" ]] || false
+}
+
+@test "keyless diff branches with identical mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on other"
+
+    run dolt diff master
+    [ $status -eq 0 ]
+    [ "$output" = "" ]
+}
+
+@test "keyless merge branches with identical mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on other"
+
+    run dolt merge master
+    [ $status -eq 0 ]
+    run dolt sql -q "SELECT * FROM keyless WHERE c0 > 6;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
```

Comment:
Is this supposed to be checking that there is only one copy of each row? It's not testing that